From 25c6c235b65bf0a19fd37ca05aac1d1214c7cc7c Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Mon, 11 Feb 2013 15:24:49 +0200
Subject: [PORT FROM MAIN] Telephony: Fix issue with overlapping stall alarms

BZ: 68905

When the data connection is disconnected due to NW DETACH, the
data stall alarm is not stopped. If the deactivation of context
after NW DETACH takes a longer time and there are active applications
trying sending and receiving, the data stall alarm is possible.
This alarm might disconnect an active connection if there is any
established after the deactivate of the first connection.

This patch stops the data stall alarm, if there are no data connections.

Change-Id: I6aaa9b35d9a93a9795ff7cdeef7b3de587648205
Orig-Change-Id: I79016c755dc9b2ec93b2e19e36e0a8cd3a88a455
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
Reviewed-on: http://android.intel.com:8080/91293
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Lucas, Guillaume <guillaume.lucas@intel.com>
Reviewed-by: Badrappan, Jeevaka <jeevaka.badrappan@intel.com>
Reviewed-by: D'alton, Alexandre <alexandre.d'alton@intel.com>
Reviewed-by: Fremaux, Pascal <pascal.fremaux@intel.com>
Reviewed-by: Beare, Bruce J <bruce.j.beare@intel.com>
Reviewed-by: Lebsir, SamiX <samix.lebsir@intel.com>
Tested-by: Lebsir, SamiX <samix.lebsir@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 .../telephony/dataconnection/DcTracker.java        |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
index bbe10eb..f5fb55b 100644
--- a/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
+++ b/src/java/com/android/internal/telephony/dataconnection/DcTracker.java
@@ -1133,6 +1133,12 @@ public final class DcTracker extends DcTrackerBase {
             if (isAnyDataCallActive) startNetStatPoll();
         }
 
+        if (dataCallStates.size() == 0) {
+            if (DBG) log("onDataStateChange(ar): NoData calls Stop stall alarm");
+            stopNetStatPoll();
+            stopDataStallAlarm();
+        }
+
         if (DBG) log("onDataStateChanged(ar): X");
     }
 
-- 
1.7.4.1

